corgi/core/models.py

```diff
@@ -306,10 +306,12 @@ def save_product_taxonomy(self):
         for component in self.components.get_queryset():
             # This is needed for container image builds which pull in components not
             # built at Red Hat, and therefore not assigned a build_id
-            for d in component.cnodes.get_queryset().get_descendants(include_self=True):
-                if not d.obj:
-                    continue
-                components.add(d.obj)
+            for cnode in component.cnodes.get_queryset().prefetch_related("obj"):
+                components.add(cnode.obj)
+                for d in cnode.get_descendants():
+                    if not d.obj:
+                        continue
+                    components.add(d.obj)
```

Comment:
@jasinner I think pretetch_related here will slow down the query (more data transferred), but the obj is only accessed in a different query so we're not using it here (usually).

Try putting it on the inner query, like below:
```suggestion
            for cnode in component.cnodes.get_queryset():
                for d in cnode.get_descendants(include_self=True).prefetch_related("obj"):
                    if not d.obj:
                        continue
                    components.add(d.obj)
```

Reply:
I tried it and it was slower for Brew build 1875516.